histomicstk/cli/SuperpixelSegmentation/SuperpixelSegmentation.xml

```diff
@@ -0,0 +1,167 @@
+  <documentation-url>https://digitalslidearchive.github.io/HistomicsTK/</documentation-url>
+  <license>Apache 2.0</license>
+  <contributor>Kitware</contributor>
+  <acknowledgements>This work is part of the HistomicsTK project.</acknowledgements>
+  <parameters>
+    <label>IO</label>
+    <description>Input/output parameters.</description>
+    <image>
+        <name>inputImageFile</name>
+        <label>Input Image</label>
+        <description>Input image for superpixel segmentation</description>
+        <channel>input</channel>
+        <index>0</index>
+    </image>
+    <image fileExtensions=".tiff" reference="inputImageFile">
+      <name>outputImageFile</name>
+      <label>Output Image of Superpixel Pixelmap</label>
+      <description>Output Image of Superpixel Pixelmap (*.tiff)</description>
+      <channel>output</channel>
+      <index>1</index>
+    </image>
+    <file fileExtensions=".anot" reference="inputImageFile">
+      <name>outputAnnotationFile</name>
+      <label>Pixelmap Annotation</label>
+      <description>Annotation to display pixelmap on source (*.anot)</description>
+      <channel>output</channel>
+      <longflag>superpixel_annotation</longflag>
+    </file>
+    <region>
+      <name>roi</name>
+      <label>Analysis ROI</label>
+      <longflag>roi</longflag>
+      <description>Region of interest within which the analysis should be run</description>
+      <channel>input</channel>
+      <default>-1, -1, -1, -1</default>
+    </region>
+    <integer>
+       <name>tileSize</name>
+       <longflag>tileSize</longflag>
+       <label>Working tile size</label>
+       <description>Specify the size of the working tile. If there is no overlap, superpixxel boundaries will appear at these locations.</description>
+       <channel>input</channel>
+       <default>4096</default>
+    </integer>
+    <integer>
+       <name>superpixelSize</name>
+       <longflag>superpixelSize</longflag>
+       <label>Superpixel Size</label>
+       <description>Approximate width of the superpixel.</description>
+       <channel>input</channel>
+       <default>50</default>
+    </integer>
+    <boolean>
+      <name>overlap</name>
+      <label>Overlap</label>
+      <longflag>overlap</longflag>
+      <description>If specified, overlap tile computation to avoid edge effects.</description>
+      <channel>input</channel>
+      <default>false</default>
+    </boolean>
+    <boolean>
+      <name>boundaries</name>
+      <label>Boundaries</label>
+      <longflag>boundaries</longflag>
+      <description>If specified, draw separate superpixels as boundaries around actual superpixels.</description>
+      <channel>input</channel>
+      <default>true</default>
+    </boolean>
+    <float>
+      <name>compactness</name>
+      <label>Compactness</label>
+      <longflag>compactness</longflag>
+      <description>SLIC Algorithm Compactness</description>
+      <channel>input</channel>
+      <default>1.0</default>
```

Comment:
skimage defaults to a compactness of 10 and sigma of 0.  We should probably ask @cooperlab what defaults are actually good for histology images.  Visually, a compactness of 0.25 looks better to me, but that is based more on personal aesthetics than on algorithm utility.

Reply:
@cooperlab Do you have any opinions on this versus the "slic-zero" variant of the algorithm?  We could expose both by using a compactness of 0 to invoke the slic-zero algorithm (if so, internally, we'd have to not pass compactness or use a value of 1 or skimage will fail).

Reply:
I recommend defaulting to slic-zero. People who know what compatedness means can fool with the parameters. A good compactedness parameter depends entirely on the tissue type.

Reply:
I've updated the parameters so that the default value for `compactness` is `0`, and the algorithm will interpret that as a cue to use SLICO. I am noticing that this results in very boxy superpixels on my test images. b032305

Reply:
Looking at skimage source, the compactness value is still used with SLIC0.  Specifically, the image is scaled to full dynamic range and then to 1/compactness before being passed to the SLIC or SLIC0 algorithm.  In a quick experiment, the values DO matter, with a compactness of 0.1 following image features much better than the default (10).

I think we need to go back to having a boolean flag for SLIC0, and default the compactness to 1 and the sigma to 0.

Reply:
I've made the suggested changes and can verify that appears to bean improvement over using `compactness = 0` to invoke SLIC0.
2b71db4